proposals/NNNN-library-evolution.md

```diff
@@ -0,0 +1,374 @@
+
+  public init(_ x: Double, _ y: Double) {
+    self.x = x
+    self.y = y
+  }
+}
+```
+
+This affects what changes to the struct's fields affect the ABI of the containing library:
+
+| Change | Normal struct | `@frozen` struct
+|---|:---:|:---:
+| Adding fields | Allowed | **Affects ABI**
+| Reordering fields | Allowed | **Affects ABI**
+| Removing ABI-public fields | Affects ABI | Affects ABI
+| Removing non-ABI-public fields | Allowed | **Affects ABI**
+| Changing the type of an ABI-public field | Affects ABI | Affects ABI
+| Changing the type of a non-ABI-public field | Allowed | **Affects ABI**
+| Changing a stored instance property to computed | Allowed | **Affects ABI**
+| Changing a computed instance property to stored | Allowed | **Affects ABI**
+| Changing the access of a non-ABI-public field | Allowed | Allowed
+| Marking an `internal` field as `@abiPublic` | Allowed | Allowed
+| Changing an `internal` ABI-public field to be `public` | Allowed | Allowed
+
+
+> Note: This proposal is implemented already and in use by the standard library, albeit under different names. The command-line flag is `-enable-resilience`; the attribute is `@_fixed_layout` for structs, and `@frozen` for enums.
+
+  [SE-0193]: https://github.com/apple/swift-evolution/blob/master/proposals/0193-cross-module-inlining-and-specialization.md
+
+#### Guarantees
+
+Marking a struct `@frozen` only guarantees that its stored instance properties won't change. This allows the compiler to perform certain optimizations, like ignoring properties that are never accessed, or eliminating redundant loads from the same instance property. However, it does not provide a handful of other guarantees that a C struct might:
+
+- **It is not guaranteed to be "trivial"** ([in the C++ sense][trivial]). A frozen struct containing a class reference or closure still requires reference-counting when copied and when it goes out of scope.
+
+- **It is not guaranteed to be stored in registers.** Promoting structs to be stored in registers is considered an optimization in Swift, and as such the compiler is never *required* to do so. In particular, a struct with a `weak` reference in it must always be stored in memory if the compiler cannot prove that the weak reference is unused.
+
+- **It does not necessarily have a known size or alignment.** A generic frozen struct's layout might depend on the generic argument provided at run time.
+
+- **Even concrete instantiations may not have a known size or alignment.** A frozen struct with a field that's a *non-*frozen, has a size that may not be known until run time.
+
+- **It is not guaranteed to use the same layout as a C struct with a similar "shape".** If such a struct is necessary, it should be defined in a C header and imported into Swift.
+
+- **The fields are not guaranteed to be laid out in declaration order.** The compiler may choose to reorder fields to minimize padding while satisfying alignment requirements, for example, .
```

Comment:
"for example"?

Reply:
Ah, the example is preceding, I just failed to trim a comma from a removed clause.